### PR TITLE
[form-data] Remove @types/form-data since v2.5.0 it bundles types

### PR DESCRIPTION
**Pull request by mapleeit**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: notNeededPackages.json [hunk 1]
```diff
@@ -1080,6 +1080,12 @@
             "sourceRepoURL": "https://github.com/Realytics/fork-ts-checker-webpack-plugin",
             "asOfVersion": "0.4.5"
         },
+        {
+            "libraryName": "form-data",
+            "typingsPackageName": "form-data",
+            "sourceRepoURL": "https://github.com/form-data/form-data",
+            "asOfVersion": "2.5.0"
+        },
```
Comment (ffflorian) [inline notNeededPackages.json]: This already exists in line 1072.

**Review by ffflorian — CHANGES_REQUESTED** [review]

File: notNeededPackages.json (hunk 1 above)
Comment (mapleeit) [reply notNeededPackages.json]: ops...Didn't notice that. I'll delete this.

**Review by ffflorian — APPROVED** [review]

**Review by LinusU — APPROVED** [review]

**Review by sandersn — CHANGES_REQUESTED** [review]
I think `got` builds without the dependency. At least it did [in the overnight run](https://dev.azure.com/definitelytyped/DefinitelyTyped/_build/results?buildId=9026). So can you verify that this change is needed?

**Comment by mapleeit** [comment]
@sandersn I'm not sure about this. I just followed tutorial of removing a package on readme. https://github.com/DefinitelyTyped/DefinitelyTyped#removing-a-package

I noticed that `got` have a nested node_modules. And outer layer contains the dependencies declaration. I'm not sure should inner contains or not.

image

**Comment by sandersn** [comment]
A couple of points.
1. #36819 already removed form-data, as far as I can tell.
2. The instructions say to add dependencies on everything that fails with `npm test` after form-data is removed. Does got fail after form-data is removed but before the dependency is added to package.json?

**Comment by mapleeit** [comment]
Got it. I'll close this.

**Pull request closed without merging**